### PR TITLE
Change to use dockerize when starting up container, and fix deprecate…

### PR DESCRIPTION
**Pull request by sychan**
…d maintainer

Requires updated kb_minideb container that includes dockerize binary.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -18,6 +17,9 @@ LABEL org.label-schema.build-date=$BUILD_DATE \
       org.label-schema.vcs-url="https://github.com/kbase/Shock.git" \
       org.label-schema.vcs-ref=$VCS_REF \
       org.label-schema.schema-version="1.0.0-rc1" \
-      us.kbase.vcs-branch=$BRANCH
+      us.kbase.vcs-branch=$BRANCH \
+      maintainer="Steve Chan sychan@lbl.gov"
 
-ENTRYPOINT [ "/kb/deployment/bin/entrypoint.sh" ]
+ENTRYPOINT [ "/kb/deployment/bin/dockerize"]
```
Comment (kkellerlbl) [inline Dockerfile:23]: So dockerize completely replaces the entrypoint script, and everything that it calls is put into docker-compose.yml?

File: docker-compose.yml [hunk 2]
```diff
@@ -2,6 +2,16 @@ version: "3.1"
 services:
   shock:
     image: kbase/kb_shock:${TRAVIS_BRANCH:-latest}
+    command:
+      - "-template"
+      - "/kb/deployment/conf/.templates/shock-server.cfg.templ:/kb/deployment/conf/shock-server.cfg"
+      - "-wait"
+      - "tcp://mongo:27017"
```
Comment (kkellerlbl) [inline docker-compose.yml:9]: Is this where dockerize waits for the target to be up before continuing?

**Review by kkellerlbl — APPROVED** [review]
Looks ok to me if it works

File: Dockerfile (hunk 1 above)
Comment (sychan) [reply Dockerfile:23]: Yep! The entrypoint script and it's dependencies ( python, wget ) are all gone, trimming 30M off the image size. Even after extending dockerize to pull data from remote INI files, it only expanded the dockerize binary by a couple of megs.

File: docker-compose.yml (hunk 2 above)
Comment (sychan) [reply docker-compose.yml:9]: Yes, also look at the next "-wait" parameter that checks to make sure that the mongoinit container completed loading into mongodb before trying to spin up Shock. Before there was a race condition between mongoinit and Shock as to whether mongorestore would complete before Shock tried to use it.

File: docker-compose.yml [hunk 3]
```diff
@@ -2,13 +2,35 @@ version: "3.1"
...
+      - "150s"
+      - "/kb/deployment/bin/shock-server"
+      - "--conf"
+      - "/kb/deployment/conf/shock-server.cfg"
     ports:
       - "7044:7044"
+    depends_on: ['mongoinit',"mongo"]
   mongoinit:
-    image: mongo:2
+    image: kbase/db_initialize:latest
```
Comment (sychan) [inline docker-compose.yml:21]: @kkellerlbl This is a new image that specializes in loading databases and has a little wrapper script that brings up a netcat listener after loading the data, which a dependent container can check as a sign that the data load completed.

File: docker-compose.yml (hunk 2 above)
Comment (kkellerlbl) [reply docker-compose.yml:9]: That's pretty nifty.  What's on port 8080 in the mongoinit container?

File: docker-compose.yml (hunk 3 above)
Comment (kkellerlbl) [reply docker-compose.yml:21]: We may want to tell people who want to preserve an existing environment (either an existing *.kbase.us environment, or maybe someone wants their local minikb persistent) not to use the mongoinit container after the initial init.

**Pull request merged**